generate-release-notes.sh

```diff
@@ -90,94 +130,205 @@ release_notes="### Source AMI release notes
 ### Changelog
 ---
 https://github.com/aws/amazon-ecs-ami/blob/main/CHANGELOG.md#$ami_version
+"
+
+# AL2023
+if ! { is_ami_excluded "al2023" && is_ami_excluded "al2023arm" && is_ami_excluded "al2023neu"; }; then
+    # Get AL2023 AMI family details
+    readonly containerd_version_al2023=$(cat $variablespkr | sed -n '/containerd_version_al2023"/,/}/p' | grep -w 'default' | cut -d '"' -f2)
+    readonly distribution_release_al2023=$(cat $al2023pkrvars | grep -w 'distribution_release_al2023' | cut -d '"' -f2)
+    if [ -z "$containerd_version_al2023" ]; then
+        echo "Error: Containerd version was not found for AL2023 in $al2023pkrvars"
+        exit 1
+    fi
+    if [ -z "$distribution_release_al2023" ]; then
+        echo "Error: Distribution release version was not found for AL2023 in $al2023pkrvars"
+        exit 1
+    fi
 
+    # AL2023 Header
+    al2023_header="
 ### Amazon ECS-optimized Amazon Linux 2023 AMI
----
+---"
+    release_notes="${release_notes}${al2023_header}"
+
+    # Include AL2023 AMD64 release notes if there was an al2023 release
+    if ! is_ami_excluded "al2023"; then
+        # AL2023 AMD64 AMI details
+        read ami_name_al2023_x86 agent_version_al2023_x86 docker_version_al2023_x86 source_ami_name_al2023_x86 <<<$(get_ami_details "/aws/service/ecs/optimized-ami/amazon-linux-2023/recommended")
+        al2023_release_notes="
```

Comment:
not & non-blocking. I think we can have a common function to generate `alx_release_notes` for each ECS-optimized AMI flavor in a follow-up CR.

```
#### $ami_arch
- AMI name: $ami_name
- ECS Agent version: [$agent_version](https://github.com/aws/amazon-ecs-agent/releases/tag/v$agent_version)
- Docker version: $docker_version
- Containerd version: $containerd_version
- Source AMI name: $source_ami_name
- Distribution release: $distribution_release
```

Reply:
This is a great point / idea! I will add this in.